### PR TITLE
fixed order by expression

### PR DESCRIPTION
**Pull request by SamelVhatargh**

### CODE REVIEW

File: tests/Query/SelectTest.php [hunk 1]
```diff
@@ -254,6 +255,17 @@ public function testOrderByReset()
         $this->assertParams([], $select);
     }
 
+    public function testOrderByExpression()
+    {
+        $select = $this->factory
+            ->select()
+            ->from('users')
+            ->orderBy(express("FIELD(`status`, 'off')"), 'DESC');
```
Comment (shadowhand) [inline tests/Query/SelectTest.php]: This is a bad example, as the `status` is quoted in a way that is specific to MySQL.

Comment (shadowhand) [inline tests/Query/SelectTest.php]:
```suggestion
            ->orderBy(express("FIELD(%s, 'off')", "status"), 'DESC');
```

File: tests/Query/SelectTest.php [hunk 2]
```diff
@@ -254,6 +255,17 @@ public function testOrderByReset()
...
     }
 
+    public function testOrderByExpression()
+    {
+        $select = $this->factory
+            ->select()
+            ->from('users')
+            ->orderBy(express("FIELD(%s, 'off')", 'status'), 'DESC');
+
+        $this->assertSql("SELECT * FROM users ORDER BY FIELD(?, 'off') DESC", $select);
```
Comment (shadowhand) [inline tests/Query/SelectTest.php]: Hmm, this isn't right. The SQL should have the field in the query, not a parameter.

Bad suggestion in my last review.

Changing `express(..., 'status')` to `express(..., identify('status'))` should do it.

Or, perhaps a more simple test that accomplishes the same thing:

```php
$select = $this->factory->select()->from('users')->orderBy(literal('RAND()'));
```

**Comment by SamelVhatargh** [comment]
Are you asking me to do something like
`$ git rebase -i`
`$ git push origin --force`
?

Maybe it will be better to use "squash and merge" option when merging pull request?
https://help.github.com/articles/about-pull-request-merges/#squash-and-merge-your-pull-request-commits

**Comment by shadowhand** [comment]
@SamelVhatargh sure, I can do a squash and merge if that is easier. Thanks!

**Comment by shadowhand** [comment]
I'll do a release with this change in the next couple of days.

**Comment by SamelVhatargh** [comment]
Thanks!

**Pull request merged**